scripts/spack/packages/spheral/package.py

```diff
@@ -67,8 +67,11 @@ class Spheral(CachedCMakePackage, CudaPackage, ROCmPackage):
     depends_on('conduit@0.9.1 +shared +hdf5~hdf5_compat -test ~parmetis', type='build')
 
     depends_on('axom@0.9.0 +hdf5 -lua -examples -python -fortran', type='build')
-    depends_on('axom +shared', when='~cuda', type='build')
-    depends_on('axom ~shared', when='+cuda', type='build')
+    with when('+rocm') or when('+cuda'):
+        depends_on('axom ~shared', type='build')
+
+    with when('~rocm') or when('~cuda'):
+        depends_on('axom +shared', type='build')
 
     with when("+caliper"):
```

Comment:
@ldowen do you have a preferred way to do this, my solution here seems subpar...

Reply:
I couldn't find a better way to do it. I think I would prefer listing them all out with `when` statements instead of using the `with when` but how you have it is fine if it works.